packages/arb-token-bridge-ui/src/components/RecoverFunds.tsx

```diff
@@ -0,0 +1,560 @@
+import {
+  Address,
+  EthBridger,
+  getArbitrumNetwork,
+  ParentToChildMessageGasEstimator,
+  ParentTransactionReceipt
+} from '@arbitrum/sdk'
+import useSWRImmutable from 'swr/immutable'
+import {
+  getNetworkName,
+  getSupportedChainIds,
+  isNetwork
+} from '../util/networks'
+import { getProviderForChainId } from '@/token-bridge-sdk/utils'
+import { BigNumber, constants, Signer } from 'ethers'
+import { useNetworks } from '../hooks/useNetworks'
+import { useAccount } from 'wagmi'
+import { useMemo, useState } from 'react'
+import { fetchNativeCurrency, NativeCurrency } from '../hooks/useNativeCurrency'
+import { ChainId } from '../types/ChainId'
+import { formatAmount } from '../util/NumberUtils'
+import { isAddress, parseEther } from 'ethers/lib/utils'
+import { errorToast } from './common/atoms/Toast'
+import { getBaseFee } from '@arbitrum/sdk/dist/lib/utils/lib'
+import { StaticJsonRpcProvider } from '@ethersproject/providers'
+import { Inbox__factory } from '@arbitrum/sdk/dist/lib/abi/factories/Inbox__factory'
+import { ERC20Inbox__factory } from '@arbitrum/sdk/dist/lib/abi/factories/ERC20Inbox__factory'
+import { ParentToChildMessageGasParams } from '@arbitrum/sdk/dist/lib/message/ParentToChildMessageCreator'
+import { Button } from './common/Button'
+import { getInboxAddressFromOrbitChainId } from '../util/orbitChainsList'
+import { useEthersSigner } from '../util/wagmi/useEthersSigner'
+import { isUserRejectedError } from '../util/isUserRejectedError'
+import { useError } from '../hooks/useError'
+import { Column, Table, TableCellRenderer } from 'react-virtualized'
+import { DialogWrapper, useDialog2 } from './common/Dialog2'
+import { Dialog, UseDialogProps } from './common/Dialog'
+import { SafeImage } from './common/SafeImage'
+import { NetworkImage } from './common/NetworkImage'
+import { twMerge } from 'tailwind-merge'
+import { Loader } from './common/atoms/Loader'
+import { NoteBox } from './common/NoteBox'
+import { trackEvent } from '../util/AnalyticsUtils'
+import { shortenAddress } from '../util/CommonUtils'
+import { Tooltip } from './common/Tooltip'
+import { TokenLogoFallback } from './TransferPanel/TokenInfo'
+import { addressesEqual } from '../util/AddressUtils'
+import { useSwitchNetworkWithConfig } from '../hooks/useSwitchNetworkWithConfig'
+import { useLatest } from 'react-use'
+
+async function createRetryableTicket({
```

Comment:
NIT: I would rename this to `createRetryableTicketForFundsRecovery()` or `performFundRecoveryTransaction()` being more specific to the use-case since it confused me a bit when internally we are using `unsafeCreateRetryableTicket`